docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,131 @@
+# Adding code beyond starter files
+
+After you [create a Kedro project](../get_started/new_project.md) and
+[add a pipeline](../tutorial/create_a_pipeline.md), you notice that Kedro generates a
+few boilerplate files: `nodes.py`, `pipeline.py`, `pipeline_registry.py`...
+
+While those may be sufficient for a small project, they quickly become large, hard to
+read and collaborate on as your codebase grows.
+Those files also sometimes make new users think that Kedro requires code
+to be located only in those starter files, which is not true.
+
+This section elaborates what are the Kedro requirements in terms of organising code
+in files and modules.
+It also provides examples of common scenarios such as sharing utilities between
+pipelines and using Kedro in a monorepo setup.
+
+## Where does Kedro look for code to be located
+
+The only technical constraint for arranging code in the project is that `pipeline_registry.py`
+file must be located in `<your_project>/src/<your_project>` directory, which is where
+it is created by default.
+
+This file must have a `register_pipelines()` function that returns a `tp.Dict[str, Pipeline]`
+mapping from pipeline name to corresponding `Pipeline` object.
+
+Other than that, **Kedro does not impose any constraints on where you should keep files with
+`Pipeline`s, `Node`s, or functions wrapped by `node`**.
+
+```{note}
+You actually can make Kedro look for pipeline registry in a different place by modifying the
+`__main__.py` file of your project, but such advanced customizations are not in scope of this article.
+```
+
+This being the only constraint means that you can, for example:
```

Comment:
It would be great if the points below are linked to the section that describe them in more detail. A bit like how we setup the configuration page with "how to.." https://kedro--3852.org.readthedocs.build/en/3852/configuration/configuration_basics.html#how-to-use-kedro-configuration

Reply:
Thanks Merel, great suggestion, just implemented this. Rendered build [here](https://kedro--3852.org.readthedocs.build/en/3852/kedro_project_setup/code_beyond_starter_files.html).